services/polly/src/main/java/software/amazon/awssdk/services/polly/internal/presigner/DefaultPollyPresigner.java

```diff
@@ -233,10 +316,13 @@ private AwsCredentialsIdentity resolveCredentials(IdentityProvider<? extends Aws
 
     private Presigner resolvePresigner(PollyRequest request) {
         Signer signer = request.overrideConfiguration().flatMap(AwsRequestOverrideConfiguration::signer)
-                .orElse(DEFAULT_SIGNER);
-
+                               .orElse(null);
+        if (signer == null) {
+            return null;
+        }
         return Validate.isInstanceOf(Presigner.class, signer,
-                "Signer of type %s given in request override is not a Presigner", signer.getClass().getSimpleName());
+                                     "Signer of type %s given in request override is not a Presigner",
+                                     signer.getClass().getSimpleName());
```

Comment:
I understand this is existing code, but getSimpleName is quite expensive, can we change to just use getName() while we are at it?

Reply:
I didn't know that, now I do, I will update to use `getName()` instead.